src/library-authoring/components/ComponentCard.test.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import { AppProvider } from '@edx/frontend-platform/react';
+import { initializeMockApp } from '@edx/frontend-platform';
+import { getAuthenticatedHttpClient } from '@edx/frontend-platform/auth';
+import { IntlProvider } from '@edx/frontend-platform/i18n';
+import { render, fireEvent, screen } from '@testing-library/react';
+import MockAdapter from 'axios-mock-adapter';
+import type { Store } from 'redux';
+
+import { ToastProvider } from '../../generic/toast-context';
+import { ContentHit } from '../../search-manager';
+import initializeStore from '../../store';
+import ComponentCard from './ComponentCard';
+
+let store: Store;
+let axiosMock: MockAdapter;
+
+const contentHit: ContentHit = {
+  id: '1',
+  usageKey: 'lb:org1:demolib:html:a1fa8bdd-dc67-4976-9bf5-0ea75a9bca3d',
+  type: 'library_block',
+  blockId: 'a1fa8bdd-dc67-4976-9bf5-0ea75a9bca3d',
+  contextKey: 'lb:org1:Demo_Course',
+  org: 'org1',
+  breadcrumbs: [{ displayName: 'Demo Lib' }],
+  displayName: 'Text Display Name',
+  formatted: {
+    displayName: 'Text Display Formated Name',
+    content: {
+      htmlContent: 'This is a text: ID=1',
+    },
+  },
+  tags: {
+    level0: ['1', '2', '3'],
+  },
+  blockType: 'text',
+  created: 1722434322294,
+  modified: 1722434322294,
+  lastPublished: null,
+};
+
+const RootWrapper = () => (
+  <AppProvider store={store}>
+    <IntlProvider locale="en">
+      <ToastProvider>
+        <ComponentCard
+          contentHit={contentHit}
+          blockTypeDisplayName="text"
+        />
+      </ToastProvider>
+    </IntlProvider>
+  </AppProvider>
+);
+
+describe('<ComponentCard />', () => {
+  beforeEach(() => {
+    initializeMockApp({
+      authenticatedUser: {
+        userId: 3,
+        username: 'abc123',
+        administrator: true,
+        roles: [],
+      },
+    });
+    store = initializeStore();
+
+    axiosMock = new MockAdapter(getAuthenticatedHttpClient());
+  });
+
+  afterEach(() => {
+    jest.resetAllMocks();
+    axiosMock.restore();
+  });
+
+  it('should render the card with title and description', () => {
+    const { getByText } = render(<RootWrapper />);
+
+    expect(getByText('Text Display Formated Name')).toBeInTheDocument();
+    expect(getByText('This is a text: ID=1')).toBeInTheDocument();
+  });
+
+  it('should call the updateClipboard function when the copy button is clicked', () => {
+    axiosMock.onPost('/api/clipboard/').reply(200, {});
+    const { getByTestId } = render(<RootWrapper />);
+
+    // Open menu
+    fireEvent.click(getByTestId('component-card-menu-toggle'));
+    fireEvent.click(getByTestId('component-card-menu-copy-clipboard'));
+
```

Comment:
```suggestion
    // Open menu
    await waitFor(() => {
      expect(getByTestId('component-card-menu-toggle')).toBeInTheDocument();
    });
    fireEvent.click(getByTestId('component-card-menu-toggle'));

    // Copy to clipboard
    await waitFor(() => {
      expect(getByTestId('component-card-menu-copy-clipboard')).toBeInTheDocument();
    });
    fireEvent.click(getByTestId('component-card-menu-copy-clipboard'));
```
(dont forget to import `waitFor` at the beginning of the page)

@rpenido I figured out a fix for the issue with the tests, the error was not very descriptive. It was failing because the toggle dropdown was not rendered on the page before we attempt to click on it, similarly we needed to wait for the copy to clipboard button to appear on the screen to be able to click on it.

Reply:
Hi @yusuf-musleh! Thank you for your help, but I still have the same problem. Can you confirm that this was the only change and the test passed?
```
 FAIL  src/library-authoring/components/ComponentCard.test.tsx (27.273 s)
  <ComponentCard />
    ✓ should render the card with title and description (59 ms)
    ✕ should call the updateClipboard function when the copy button is clicked (1166 ms)

  ● <ComponentCard /> › should call the updateClipboard function when the copy button is clicked

    Unable to find an element by: [data-testid="component-card-menu-copy-clipboard"]

    Ignored nodes: comments, script, style
    <body>
      <div>
        <div
          class="py-5 justify-content-center align-items-start text-center container-fluid"
          data-testid="error-page"
        >
          <div
            class="row"
          >
            <div
              class="col"
            >
              <p
                class="text-muted"
              >
                An unexpected error occurred. Please click the button below to refresh the page.
              </p>
              <button
                class="btn btn-primary"
                type="button"
              >
                Try again
              </button>
            </div>
          </div>
        </div>
      </div>
    </body>

      90 |     fireEvent.click(getByTestId('component-card-menu-toggle'));
      91 |
    > 92 |     await waitFor(() => {
         |                  ^
      93 |       expect(getByTestId('component-card-menu-copy-clipboard')).toBeInTheDocument();
      94 |     });
      95 |     fireEvent.click(getByTestId('component-card-menu-copy-clipboard'));

      at waitForWrapper (node_modules/@testing-library/react/node_modules/@testing-library/dom/dist/wait-for.js:166:27)
      at src/library-authoring/components/ComponentCard.test.tsx:92:18
      at fulfilled (src/library-authoring/components/ComponentCard.test.tsx:5:58)
```

Reply:
@rpenido I double checked again, you're right the code does still error with just these changes, when I tested it the first time, I commented out the other test case, when I run them together I get this error, so it seems like it is affecting the other test.

I managed to fix it by swapping out `jest.resetAllMocks();` -> `jest.clearAllMocks();` in the `afterEach`, now all the tests should start passing, with the change from above.

Can you double check again to confirm?

Reply:
Awesome @yusuf-musleh! It worked!!! Thank you!